src/shapes/Group.spec.ts

```diff
@@ -338,4 +410,1396 @@ describe('Group', () => {
+      'a-canvas',
+    );
+  });
+
+  it('canvas prop propagation with add', () => {
+    const rect1 = new Rect({
+        top: 1,
+        left: 1,
+        width: 2,
+        height: 2,
+        strokeWidth: 0,
+        fill: 'red',
+        opacity: 1,
+        objectCaching: true,
+      }),
+      rect2 = new Rect({
+        top: 5,
+        left: 5,
+        width: 2,
+        height: 2,
+        strokeWidth: 0,
+        fill: 'red',
+        opacity: 1,
+        objectCaching: true,
+      }),
+      group = new Group([rect1, rect2]);
+
+    canvas.add(group);
+    expect(group.canvas, 'canvas has been set').toBe(canvas);
+    expect(group._objects[0].canvas, 'canvas has been set on object 0').toBe(
+      canvas,
+    );
+    expect(group._objects[1].canvas, 'canvas has been set on object 1').toBe(
+      canvas,
+    );
+  });
+
+  it('canvas prop propagation with add to group', () => {
+    const rect1 = new Rect({
+        top: 1,
+        left: 1,
+        width: 2,
+        height: 2,
+        strokeWidth: 0,
+        fill: 'red',
+        opacity: 1,
+        objectCaching: true,
+      }),
+      rect2 = new Rect({
+        top: 5,
+        left: 5,
+        width: 2,
+        height: 2,
+        strokeWidth: 0,
+        fill: 'red',
+        opacity: 1,
+        objectCaching: true,
+      }),
+      group = new Group();
+
+    canvas.add(group);
+    expect(group.canvas, 'canvas has been set').toBe(canvas);
+    group.add(rect1);
+    expect(group._objects[0].canvas, 'canvas has been set on object 0').toBe(
+      canvas,
+    );
+    group.add(rect2);
+    expect(group._objects[1].canvas, 'canvas has been set on object 0').toBe(
+      canvas,
+    );
+  });
+
+  it('add and coordinates', () => {
+    const rect1 = new Rect({
+        top: 1,
+        left: 1,
+        width: 3,
+        height: 2,
+        strokeWidth: 0,
+        fill: 'red',
+      }),
+      rect2 = new Rect({
+        top: 5,
+        left: 5,
+        width: 2,
+        height: 6,
+        angle: 90,
+        strokeWidth: 0,
+        fill: 'red',
+      }),
+      group = new Group([], { layoutManager: new LayoutManager() });
+
+    group.add(rect1);
+    group.add(rect2);
+    group.left = 5;
+    group.top = 5;
+    group.scaleX = 3;
+    group.scaleY = 2;
+    group.removeAll();
+
+    expect(rect1.top, 'top has been moved').toBe(5);
+    expect(rect1.left, 'left has been moved').toBe(11);
+    expect(rect1.scaleX, 'scaleX has been scaled').toBe(3);
+    expect(rect1.scaleY, 'scaleY has been scaled').toBe(2);
+    expect(rect2.top, 'top has been moved').toBe(13);
+    expect(rect2.left, 'left has been moved').toBe(23);
+    expect(
+      rect2.scaleX,
+      'scaleX has been scaled inverted because of angle 90',
+    ).toBe(2);
+    expect(
+      rect2.scaleY,
+      'scaleY has been scaled inverted because of angle 90',
+    ).toBe(3);
+  });
+
+  // QUnit.skip('addRelativeToGroup and coordinates with nested groups', function(assert) {
```

Comment:
this is a 1:1 port to vitest (with some minor adjustments for nicer matchers) and I am not sure what was the plan in qunit for this part, so I just moved this whole thing as a comment for you to decide what you want to do with it